locallib.php

```diff
@@ -1107,3 +1107,50 @@ function zoom_load_meeting($id, $context, $usestarturl = true) {
 
     return $returns;
 }
+
+/**
+ * Get the configured Zoom tracking fields.
+ *
+ * @return array tracking fields.
+ */
+function zoom_list_tracking_fields() {
+    // Get Zoom API service instance.
+    $service = new mod_zoom_webservice();
+    $trackingfields = array();
+
+    // Get the tracking fields configured on the account.
+    $response = $service->list_tracking_fields();
+    if ($response != null) {
+        foreach ($response->tracking_fields as $trackingfield) {
+            $trackingfields[] = array(
+                'id' => $trackingfield->id,
+                'field' => $trackingfield->field,
+                'required' => $trackingfield->required,
+                'visible' => $trackingfield->visible,
+                'recommended_values' => $trackingfield->recommended_values
+            );
+            $trackingfields[] = (array) $trackingfield;
+        }
+    }
+
+    return $trackingfields;
+}
+
+/**
+ * Trim and lower case tracking fields.
+ *
+ * @return array tracking fields trimmed, keys as lower case
+ */
+function zoom_clean_tracking_fields() {
+    $config = get_config('zoom');
+    $defaulttrackingfields = explode(',', $config->defaulttrackingfields);
+    $trackingfields = array();
+
+    foreach ($defaulttrackingfields as $key => $defaulttrackingfield) {
+        $trimmed = trim($defaulttrackingfield);
+        $key = strtolower($trimmed);
+        $trackingfields[$key] = $trimmed;
```

Comment:
Prevent empty keys and trackingfields:
```suggestion
        if (!empty($trimmed)) {
            $key = strtolower($trimmed);
            $trackingfields[$key] = $trimmed;
        }
```